pkg/cmd/cli/cmd/whoami.go

```diff
@@ -50,3 +54,39 @@ func (o WhoAmIOptions) WhoAmI() (*userapi.User, error) {
 
 	return me, err
 }
+
+func RunWhoAmI(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string, o *WhoAmIOptions) error {
+	if kcmdutil.GetFlagBool(cmd, "token") {
+		cfg, err := f.OpenShiftClientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		if len(cfg.BearerToken) == 0 {
+			return fmt.Errorf("no token is currently in use for this session")
+		}
+		fmt.Fprintf(out, "%s\n", cfg.BearerToken)
```

Comment:
This isn't determining if the token is valid or not.  That feel inconsistent with the base `whoami`